test/web.js

```diff
@@ -313,4 +313,70 @@ lab.experiment("OAuth", function() {
       });
     });
   });
+
+  lab.test("GET access_token - invalid grant_type", function(done) {
+    ls.start(function(error) {
+      var accessTokenRequest = {
+        method: "GET",
+        url: "/login/oauth/access_token?client_id=test&client_secret=test&grant_type=client_credentials",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded"
+        }
+      };
+
+      s.inject(authTokenRequest, function(authTokResponse) {
+        var redirectUri = url.parse(authTokResponse.headers.location, true);
+        accessTokenRequest.url += "&code=" + redirectUri.query.code;
+
+        s.inject(accessTokenRequest, function(response) {
+          Code.expect(response.statusCode).to.equal(400);
```

Comment:
I think it's also worth checking the error or message properties to make sure the correct error is being triggered